javascript/features/death_match/death_match_manager.js

```diff
@@ -175,6 +260,42 @@ export class DeathMatchManger {
 
         killer.sendMessage(Message.DEATH_MATCH_TOTAL_KILLED, killerStatistics.killCount);
         player.sendMessage(Message.DEATH_MATCH_TOTAL_DEATHS, playerStatistics.deathCount);
+
+        this.addKillToTeamForPlayer(killer);
+    }
+
+    addKillToTeamForPlayer(player) {
+        const zone = this.playersInDeathMatch_.get(player.id);
+        const playerTeam = this.playerTeam_.get(player.id);
+        const teamScore = this.teamScore_.get(zone);
+        if (zone === null || zone === undefined || playerTeam === null || playerTeam === undefined
```

Comment:
Do we really have to check for _both_ `null` and `undefined` for all of them? In most other code we use _either_ `null` or `undefined`, never both of them.

Reply:
Basically I just have trust issues in JS. You never know what might be set into the maps. (One could set a value to null instead of use delete).

Anyhow, they shouldn't and there'll be failing tests so removed the null version.